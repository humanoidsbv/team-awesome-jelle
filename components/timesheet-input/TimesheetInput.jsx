import React from 'react';

import PropTypes from 'prop-types';

import ConvertDateTime from '../../services/convert-date-time/convert-date-time';
import './timesheet-input.scss';

class TimesheetInput extends React.Component {
  static defaultFormValues = {
    employer: 'Port of Rotterdam',
    activity: 'Design',
    date: '',
    startTime: '',
    endTime: ''
  }

  static propTypes = {
    onSave: PropTypes.func.isRequired
  };

  state = {
    timeEntry: TimesheetInput.defaultFormValues,
    isFormVisible: false
  }

  toggleForm = () => {
    this.setState(({ isFormVisible }) => (
      {
        isFormVisible: !isFormVisible
      }
    ));
  }

  convertDateToUs = (date) => {
    const splittedDate = date.split('-');
    return `${splittedDate[2]}-${splittedDate[1]}-${splittedDate[0]}`;
  }

  convertToIsoString = (date, time) => new Date(`${date} ${time}`).toISOString();

  // convertDateTime = (prevState) => {
  //   const { date, startTime, endTime } = prevState.timeEntry;
  //
  //   const convertedDate = this.convertDateToUs(date);
  //
  //   const convertedStartTime = this.convertToIsoString(
  //     convertedDate, startTime.replace('.', ':')
  //   );
  //   const convertedEndTime = this.convertToIsoString(
  //     convertedDate, endTime.replace('.', ':')
  //   );
  //
  //   return {
  //     ...prevState.timeEntry,
  //     date: convertedDate,
  //     startTime: convertedStartTime,
  //     endTime: convertedEndTime
  //   };
  // }

  handleChange = ({ target }) => {
    this.setState(prevState => ({
      timeEntry: {
        ...prevState.timeEntry,
        [target.name]: target.value
      }
    }));
  };

  handleSubmit = (event) => {
    const { onSave } = this.props;
    event.preventDefault();

    const prevState = { ...this.state };
    onSave(ConvertDateTime(prevState));
    this.setState({ timeEntry: TimesheetInput.defaultFormValues });
    this.toggleForm();
  }

  render() {
    const {
      isFormVisible, timeEntry
    } = this.state;
    const {
      employer, activity, date, startTime, endTime
    } = timeEntry;

    return (
      <form>
        <div className="timesheet-input__wrapper">
          <button
            className={`
              timesheet-input__new-button
              timesheet-input__new-button${isFormVisible ? '--invisible' : '--visible'}
            `}
            onClick={this.toggleForm}
            type="button"
          >
            <img
              alt="plus sign"
              className="timesheet-field__plus-svg"
              src="/static/icons/plus.svg"
            />
            {' '}
            New time entry
          </button>
          <div className={`
            timesheet-input__form
            timesheet-input__form${isFormVisible ? '--open' : '--closed'}
            `}
          >
            <button
              className="timesheet-input__close-button"
              onClick={this.toggleForm}
              type="submit"
            />
            <div className="timesheet-input__field-item timesheet-input__field-item--employer">
              <label
                className="timesheet-input__label"
                htmlFor="employer"
                id="employer"
              >
              EMPLOYER
                <select
                  className="timesheet-input__select"
                  id="employer"
                  onChange={this.handleChange}
                  name="employer"
                  type="text"
                  value={employer}
                >
                  <option>
                    Port of Rotterdam
                  </option>
                  <option>
                    Hike One
                  </option>
                </select>
              </label>
            </div>
            <div className="timesheet-input__field-item timesheet-input__field-item--full">
              <label
                className="timesheet-input__label"
                htmlFor="activity"
                id="activity"
              >
                ACTIVITY
                <select
                  className="timesheet-input__select"
                  id="activity"
                  readOnly
                  type="text"
                  name="activity"
                  onChange={this.handleChange}
                  value={activity}
                >
                  <option>
                    Design
                  </option>
                  <option>
                    Meeting
                  </option>
                </select>
              </label>
            </div>
            <div className="timesheet-input__field-item timesheet-input__field-item--full">
              <label
                className="timesheet-input__label"
                htmlFor="date"
                id="date"
              >
                Date
                <input
                  className="timesheet-input__select"
                  id="date"
                  onChange={this.handleChange}
                  name="date"
                  type="text"
                  value={date}
                />
              </label>
            </div>
            <div className="timesheet-input__time-wrapper">
              <div className="timesheet-input__field-item timesheet-input__field-item--from">
                <label
                  className="timesheet-input__label"
                  htmlFor="from"
                >
                FROM
                  <input
                    className="timesheet-input__select"
                    id="from"
                    onChange={this.handleChange}
                    name="startTime"
                    type="text"
                    value={startTime}
                  />
                </label>
              </div>
              <div className="timesheet-input__field-item timesheet-input__field-item--to">
                <label
                  className="timesheet-input__label"
                  htmlFor="to"
                >
                  TO
                  <input
                    className="timesheet-input__select"
                    id="to"
                    onChange={this.handleChange}
                    name="endTime"
                    type="text"
                    value={endTime}
                  />
                </label>
              </div>
            </div>
            <button
              className="timesheet-input__add-button"
              type="submit"
              readOnly
              onClick={this.handleSubmit}
            >
                Add
            </button>
          </div>
        </div>
      </form>
    );
  }
}
export default TimesheetInput;
