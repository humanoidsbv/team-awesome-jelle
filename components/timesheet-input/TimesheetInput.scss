@import "../../shared/styles/main";

.timesheet-field {
  background-color: $color-accent-2;
  border: solid 1px $color-accent-3;
  border-radius: 4px;
  margin: auto;
  margin-top: 24px;
  width: 90%;

  @include desktop {
    margin-left: 12.5%;
    width: 75%;
  }

  &__field-item {
    display: grid;
    font-family: $font-primary;
    margin-left: 5%;
    margin-top: 16px;

    @include desktop {
      display: inline-grid;
    }

    &--full {
      width: 90%;
      @include desktop {
        width: 20%;
      }
    }

    &--half {
      display: inline-grid;
      width: 45%;

      @include desktop {
        display: inherit;
        width: 5%;
      }
    }

    &--last {
      display: inline-grid;
      margin-left: 0;
      width: 45%;

      @include desktop {
        display: inherit;
        margin-left: 5%;
        width: 5%;
      }
    }
  }

  &__button {
    background: $color-accent-alt-6;
    border-radius: 4px;
    color: $color-monochrome-1;
    display: grid;
    font-size: 14px;
    height: 36px;
    margin: auto;
    margin-bottom: 16px;
    margin-top: 16px;
    width: 90%;

    &--add {
      background: $color-accent-alt-6;
      //Green button for 'Add' and 'Add timesheet//
    }

    &--cancel {
      //Cross, right-handcorner when open
    }
  }

  //styling for <label> and <input> need to be added.

}

.timesheet-field-label {
  color: $color-accent-4;
  font-size: 12px;
  font-stretch: normal;
  font-style: normal;
  font-weight: 600;
  letter-spacing: normal;
  line-height: normal;
  margin-bottom: 12px;
  opacity: .5;
}

.timesheet-field-select {
  background: linear-gradient($color-accent-alt-5, $color-monochrome-1);
  border: solid 1px $color-accent-3;
  border-radius: 4px;
  font-size: 14px;
  font-style: normal;
  height: 36px;
}
