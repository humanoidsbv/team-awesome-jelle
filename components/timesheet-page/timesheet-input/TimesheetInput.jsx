import React from 'react';

import './timesheet-input.scss';

const TimesheetInput = () => (
  <div className="timesheet-field">
    <button
      className="timesheet-field__button timesheet-field__button--new"
      type="submit"
    >
      + New time entry
    </button>

    <button
      className="timesheet-field__button timesheet-field__button--close"
      type="submit"
    />
    <div className="timesheet-field__field-item timesheet-field__field-item--full">
      <label
        className="timesheet-field__label"
        htmlFor="employer"
        id="employer"
      >
        EMPLOYER
        <select
          className="timesheet-field__select"
          id="employer"
          readOnly
          type="text"
          value="Port of Rotterdam"
        >
          <option value="">
            Port of Rotterdam
          </option>
        </select>
      </label>


    </div>
    <div className="timesheet-field__field-item timesheet-field__field-item--full">
      <label
        className="timesheet-field__label"
        htmlFor="activity"
        id="activity"
      >
        ACTIVITY
        <select
          className="timesheet-field__select"
          id="activity"
          readOnly
          type="text"
          value="Design"
        >
          <option value="">
            Design
          </option>
        </select>

      </label>
    </div>
    <div className="timesheet-field__field-item timesheet-field__field-item--full">
      <label
        className="timesheet-field__label"
        htmlFor="date"
        id="date"
      >
        Date
        <select
          className="timesheet-field__select"
          id="date"
          readOnly
          type="text"
          value="29-07-2018"
        >
          <option value="">
            29-07-2018
          </option>
        </select>
      </label>
    </div>
    <div className="timesheet-field__field-item timesheet-field__field-item--half">
      <label
        className="timesheet-field__label"
        htmlFor="from"
      >
        FROM
        <select
          className="timesheet-field__select"
          id="from"
          readOnly
          type="text"
          value="09:00"
        >
          <option value="">
            09:00
          </option>
        </select>
      </label>
    </div>
    <div className="timesheet-field__field-item timesheet-field__field-item--last">
      <label
        className="timesheet-field__label"
        htmlFor="to"
      >
        TO
        <select
          className="timesheet-field__select"
          id="to"
          readOnly
          type="text"
          value="17:00"
        >
          <option value="">
            17:00
          </option>
        </select>
      </label>
    </div>
    <button
      className="timesheet-field__button timesheet-field__button--add"
      type="submit"
    >
        Add
    </button>
  </div>
);

export default TimesheetInput;
