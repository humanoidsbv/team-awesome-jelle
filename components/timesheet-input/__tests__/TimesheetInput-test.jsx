import React from 'react';
import { shallow } from 'enzyme';

import TimesheetInput from '../TimesheetInput';

test('ComponentName', () => {
  const timesheetInput = shallow(<TimesheetInput />);

  expect(timesheetInput.state()).toEqual({
    isFormVisible: false,
    timeEntry: {
      employer: 'Port of Rotterdam',
      activity: 'Design',
      date: '',
      startTime: '',
      endTime: ''
    },
    validity: {
      employer: true,
      activity: true,
      date: true,
      startTime: true,
      endTime: true
    }
  });
});
