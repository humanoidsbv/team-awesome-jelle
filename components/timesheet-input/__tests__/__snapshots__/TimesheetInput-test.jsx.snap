// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimesheetInput should render without crashing 1`] = `
<div
  className="timesheet-input__wrapper"
>
  <button
    className="
            timesheet-input__new-button
            timesheet-input__new-button--visible
          "
    onClick={[Function]}
    type="button"
  >
    <img
      alt="plus sign"
      className="timesheet-field__plus-svg"
      src="/static/icons/plus.svg"
    />
     
    New time entry
  </button>
  <button
    className="
            timesheet-input__close-button
            timesheet-input__close-button--invisible
          "
    onClick={[Function]}
    type="button"
  />
  <form
    onSubmit={[Function]}
  >
    <div
      className="
            timesheet-input__form
            timesheet-input__form--closed
            "
    >
      <div
        className="timesheet-input__field-item timesheet-input__field-item--employer"
      >
        <label
          className="timesheet-input__label"
          htmlFor="employer"
          id="employer"
        >
          EMPLOYER
          <select
            className="timesheet-input__select"
            id="employer"
            name="employer"
            onChange={[Function]}
            required={true}
            type="text"
            value="Port of Rotterdam"
          >
            <option>
              Port of Rotterdam
            </option>
            <option>
              Hike One
            </option>
          </select>
        </label>
      </div>
      <div
        className="timesheet-input__field-item timesheet-input__field-item--full"
      >
        <label
          className="timesheet-input__label"
          htmlFor="activity"
          id="activity"
        >
          ACTIVITY
          <select
            className="timesheet-input__select"
            id="activity"
            name="activity"
            onChange={[Function]}
            readOnly={true}
            required={true}
            type="text"
            value="Design"
          >
            <option>
              Design
            </option>
            <option>
              Meeting
            </option>
          </select>
        </label>
      </div>
      <div
        className="timesheet-input__field-item timesheet-input__field-item--date"
      >
        <label
          className="timesheet-input__label"
          htmlFor="date"
          id="date"
        >
          DATE
          <input
            className="
                    timesheet-input__select
                    timesheet-input__select--valid
                  "
            id="date"
            name="date"
            onBlur={[Function]}
            onChange={[Function]}
            pattern="(0[1-9]|[12][0-9]|3[01])[-](0[1-9]|1[012])[-]\\\\d{4}"
            required={true}
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        className="timesheet-input__time-wrapper"
      >
        <div
          className="timesheet-input__field-item timesheet-input__field-item--from"
        >
          <label
            className="timesheet-input__label"
            htmlFor="from"
          >
            FROM
            <input
              className="
                      timesheet-input__select
                      timesheet-input__select--valid
                    "
              id="from"
              name="startTime"
              onBlur={[Function]}
              onChange={[Function]}
              pattern="([01]?[0-9]|2[0-3]).[0-5][0-9]"
              required={true}
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          className="timesheet-input__field-item timesheet-input__field-item--to"
        >
          <label
            className="timesheet-input__label"
            htmlFor="to"
          >
            TO
            <input
              className="
                      timesheet-input__select
                      timesheet-input__select--valid
                    "
              id="to"
              name="endTime"
              onBlur={[Function]}
              onChange={[Function]}
              pattern="([01]?[0-9]|2[0-3]).[0-5][0-9]"
              required={true}
              type="text"
              value=""
            />
          </label>
        </div>
      </div>
      <button
        className="
                timesheet-input__add-button
                timesheet-input__add-button
              "
        disabled={true}
        readOnly={true}
        type="submit"
      >
        Add
      </button>
    </div>
  </form>
</div>
`;
